### ISSUE TITLE
Some Discord users join Matrix with their id instead of nickname

### ISSUE THREAD

**Issue by roughnecks** [Half-Shot/matrix-appservice-discord#870]
This is an example: @_discord_1016472849575329843_=44yno:woodpeckersnest.space

Is there a way to fix it?

**Comment by Miepee** [comment]
the `=44` part is weird, usually the format is `@_discord_[user id]:[server domain]`
Is the message from some webhook or bot?

**Comment by roughnecks** [comment]
yes, it's a bot posting a youtube video (Dyno to be exact).

**Comment by Miepee** [comment]
The bot doesn't happen to post the videos when using slash commands?

**Comment by roughnecks** [comment]
Nope, it's an automated thing. I setup my Youtube channel in bot's control panel and it sends updates to a channel whenever there's new content posted on YT.

**Comment by Miepee** [comment]
Good to know, so far I was only able to repro it with slash commands.
Am not that familiar with the discord API, but seems like there's some sort of posting from a bot, that the current bridge doesn't handle that well.
It *might* be necessary to bump the discordjs dependency in order to fix that.

**Comment by jaller94** [comment]
> the =44 part is weird

This is expected. The Matrix spec defines allowed characters in User Identifiers. In "Dyno" the upper case D is not allowed to be in a User Identifier. The bridge therefore uses URL encode to preserve the upper case D (`%44`) as described here: https://spec.matrix.org/v1.4/appendices/#mapping-from-other-character-sets

**Comment by Miepee** [comment]
Why are the display names sometimes in the MXID though?
For example here, the first one (whose MXID is `@_discord_185476724627210241:t2bot.io` was a bot command initiated with "classic" commands, while the bottom one was initiated with a slash command.
![grafik](https://user-images.githubusercontent.com/38186597/202184813-4abceb1c-eaf3-4bfb-8ba1-332e0fe3bb02.png)
